### ISSUE TITLE
[BUG] OS Integrations complain about invalid schema, invalid index names

### ISSUE THREAD

**Issue by mmehrten** [opensearch-project/dashboards-observability#1148]
**What is the bug?**
When using OpenSearch integrations, the UI will complain about invalid index names and schemas during setup even when using the same schemas that are provided in the source. For example, I used the index mappings for the CloudTrail integration [here](https://github.com/opensearch-project/dashboards-observability/blob/main/server/adaptors/integrations/__data__/repository/aws_cloudtrail/schemas/logs-aws_cloudtrail-1.0.0.mapping.json), but the UI complains about an invalid schema in the index. I also had to follow the SS4O index naming convention, but this was not documented anywhere and took trial and error to discover.

Example dataprepper pipeline: https://github.com/mmehrten/terraform/blob/main/dataprepper/docker/pipelines/pipeline.yaml

Example index templates: https://github.com/mmehrten/terraform/tree/main/dataprepper/docker/mappings/cloudtrail

Example Lambda to process CloudTrail data from CloudWatch and post to dataprepper: https://github.com/mmehrten/terraform/blob/main/dataprepper/cloudwatch_dataprepper.py

Maybe nitpicking, but I'll also call out that the provided schemas for CloudTrail don't really support CloudTrail data and will lead to ingest errors. [responseElements, userIdentity, and requestParameters](https://github.com/opensearch-project/dashboards-observability/blob/main/server/adaptors/integrations/__data__/repository/aws_cloudtrail/schemas/aws_cloudtrail-1.0.0.mapping.json#L45-L75) should all be object typed to allow more flexible schema mappings [as you can see in the sample data](https://github.com/opensearch-project/dashboards-observability/blob/main/server/adaptors/integrations/__data__/repository/aws_cloudtrail/data/samples.json#L38-L42).

**What is the expected behavior?**
CloudTrail events in SS4O format with any index name should be supported, and index validation should likely be more flexible. 

**What is your host/environment?**
AWS Managed OpenSearch 2.9.

**Do you have any screenshots?**
![image](https://github.com/opensearch-project/dashboards-observability/assets/19827255/2f6bfa50-0a5a-4e11-8bc8-d75f9cf47acf)

![image](https://github.com/opensearch-project/dashboards-observability/assets/19827255/8129a9dd-076d-4f77-9c74-ee1164b3afa5)

**Comment by derek-ho** [comment]
@Swiddis please take a look

**Comment by Swiddis** [comment]
Hi, thanks for reaching out about this. I think there are a couple integrations that are dealing with similar issues right now, I haven't had time to look into it yet. I'll have time to look at this further in a week or so, I'll be sure to keep you posted.

**Comment by Swiddis** [comment]
So, it looks like the validation issue is because of a misunderstanding in how fields are added to the mapping, which affects most integrations. I'll have a PR out soon, in the meantime here's a reproducing test:

```js
it('should correctly handle if the properties in a mapping mismatch with the mapping name', () => {
  const rootType = 'root';
  const dataSourceProps = {
    prop1: { type: 'string' },
    prop2: { type: 'number' },
  };
  const requiredMappings = {
    root: {
      template: {
        mappings: {
          properties: {
            prop1: { type: 'string' },
          },
        },
      },
    },
    child: {
      template: {
        mappings: {
          properties: {
            prop2: { type: 'number' },
          },
        },
      },
    },
  };

  const result = doPropertyValidation(rootType, dataSourceProps as any, requiredMappings);

  expect(result).toBe(true);
});
```

And another reproducer that demonstrates the false assumption:

```js
it('should not assume child templates are included as nested props', () => {
  const rootType = 'root';
  const dataSourceProps = {
    prop1: { type: 'string' },
    child: { prop2: { type: 'number' } },
  };
  const requiredMappings = {
    root: {
      template: {
        mappings: {
          properties: {
            prop1: { type: 'string' },
          },
        },
      },
    },
    child: {
      template: {
        mappings: {
          properties: {
            prop2: { type: 'number' },
          },
        },
      },
    },
  };

  const result = doPropertyValidation(rootType, dataSourceProps as any, requiredMappings);

  expect(result).toBe(false);
});
```

**Comment by Swiddis** [comment]
The bug will be fixed and deployed to 2.9 via #1234, thanks again for reaching out!

> Maybe nitpicking, but I'll also call out that the provided schemas for CloudTrail don't really support CloudTrail data and will lead to ingest errors. [responseElements, userIdentity, and requestParameters](https://github.com/opensearch-project/dashboards-observability/blob/main/server/adaptors/integrations/__data__/repository/aws_cloudtrail/schemas/aws_cloudtrail-1.0.0.mapping.json?rgh-link-date=2023-10-20T20%3A28%3A34Z#L45-L75) should all be object typed to allow more flexible schema mappings [as you can see in the sample data](https://github.com/opensearch-project/dashboards-observability/blob/main/server/adaptors/integrations/__data__/repository/aws_cloudtrail/data/samples.json?rgh-link-date=2023-10-20T20%3A28%3A34Z#L38-L42).

It looks like this fix was already merged to our [schema repository](https://github.com/opensearch-project/opensearch-catalog/blob/main/schema/observability/logs/aws/aws_cloudtrail-1.0.0.mapping) but not correctly backported to the codebase here. I'll work on taking care of that as well.

**Comment by Swiddis** [comment]
> ... though it may take a few days to hit managed service

I asked the rest of the team about it and it seems like the general flow is to require upgrades instead of backporting bugfixes. The bug isn't an issue in 2.11 since we no longer use this particular validation code (and allow adding non-naming-convention indices!), so the bug should be resolved code-wise. Still waiting for information on the schema mismatch before I mark the issue as resolved.
